controllers/cloud.redhat.com/providers/web/local.go

```diff
@@ -0,0 +1,173 @@
+package web
+
+import (
+	"fmt"
+
+	crd "github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1"
+	"github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1/common"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/config"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/errors"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/utils"
+
+	core "k8s.io/api/core/v1"
+	networking "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type mockConfig struct {
+	BOPURL      string
+	KeycloakURL string
+}
+
+type localWebProvider struct {
+	providers.Provider
+	config mockConfig
+}
+
+func NewLocalWebProvider(p *providers.Provider) (providers.ClowderProvider, error) {
+
+	wp := &localWebProvider{Provider: *p}
+
+	objList := []providers.ResourceIdent{
+		WebKeycloakDeployment,
+		WebKeycloakService,
+	}
+
+	if err := providers.CachedMakeComponent(p.Cache, objList, p.Env, "keycloak", makeKeycloak, false, p.Env.IsNodePort()); err != nil {
+		return nil, err
+	}
+
+	objList = []providers.ResourceIdent{
+		WebBOPDeployment,
+		WebBOPService,
+	}
+
+	if err := providers.CachedMakeComponent(p.Cache, objList, p.Env, "mbop", makeBOP, false, p.Env.IsNodePort()); err != nil {
+		return nil, err
+	}
+
+	wp.config.BOPURL = fmt.Sprintf("http://%s-%s.%s.svc:8080", p.Env.GetClowdName(), "mbop", p.Env.GetClowdNamespace())
+	wp.config.KeycloakURL = fmt.Sprintf("http://%s-%s.%s.svc:8080", p.Env.GetClowdName(), "keycloak", p.Env.GetClowdNamespace())
+
+	nn := types.NamespacedName{
```

Comment:
this needs to be applied in the app reconciler because it needs to be in each app ns

Reply:
done